src/main/java/software/amazon/awssdk/crt/http/HttpRequest.java

```diff
@@ -56,10 +61,52 @@ public String getEncodedPath() {
         return encodedPath;
     }
 
-    public HttpHeader[] getHeaders() {
+    public List<HttpHeader> getHeaders() {
         return headers;
     }
 
+    public void addHeader(final HttpHeader header) {
+        headers.add(header);
+    }
+
+    public void addHeader(final String headerName, final String headerValue) {
+        headers.add(new HttpHeader(headerName, headerValue));
+    }
+
+    public void addHeaders(final HttpHeader headers) {
+        this.headers.add(headers);
+    }
+
+    public void addQueryParam(final String paramName, final String paramValue, boolean urlEncode) {
```

Comment:
nuke this function if we don't need it now. no other language has this

Reply:
removed